library/CMakeLists.txt

```diff
@@ -82,6 +82,18 @@ set(MAIN_SOURCES
     RemoteTools.cpp
 )
 
+if(BUILD_TESTING)
+    # TODO Make a function or macro for this
+    add_executable(MiscUtils.test MiscUtils.test.cpp)
+    target_link_libraries(MiscUtils.test dfhack)
```

Comment:
is this necessary? what dependencies does MiscUtils have? If it does have dependencies, shouldn't they be mocked out for the unit tests? How hard is it to just build MiscUtils.cpp instead of the entire dfhack library?